src/Model/ChangeStreamIterator.php

```diff
@@ -69,9 +71,9 @@ class ChangeStreamIterator extends IteratorIterator implements CommandSubscriber
     /**
      * @internal
      * @param array|object|null $initialResumeToken
-     * @psalm-param Cursor<TValue> $cursor
+     * @psalm-param CursorInterface<int, TValue>&Iterator<int, TValue> $cursor
```

Comment:
Would you want to `assert()` on both types in the constructor? That would be temporary until we could utilize [intersection types](https://php.watch/versions/8.1/intersection-types) in 8.1+.

Reply:
Good point. I've added a check that throws an `InvalidArgumentException`.